region/tourian/main/Tourian Escape Room 4.json

```diff
@@ -269,11 +269,70 @@
           "HiJump",
           "SpaceJump",
           "canWalljump",
-          "canIBJ",
+          "canJumpIntoIBJ",
+          {"and": [
+            "canIBJ",
+            "canBePatient"
+          ]},
           "canSpringBallJumpMidAir"
+        ]},
+        {"or": [
+          "canBePatient",
+          {"enemyKill": {
+            "enemies": [
+              ["Tourian Space Pirate (all)"],
+              ["Tourian Space Pirate (all)"],
+              ["Tourian Space Pirate (all)"],
+              ["Tourian Space Pirate (all)"]
+            ],
+            "explicitWeapons": [
+              "Missile", "Super", "PowerBomb", "Wave", "Spazer", "Plasma", "ScrewAttack"
+            ]
+          }}
         ]}
       ],
-      "note": "The pirates are free to kill, although they take some time."
+      "note": ["Go up safely by killing the Pirates."]
+    },
+    {
+      "link": [1, 4],
+      "name": "Evade Pirates",
+      "requires": [
+        "canDodgeWhileShooting",
+        {"or": [
+          "HiJump",
+          "canWalljump",
+          {"and": [
+            "canTrickyDodgeEnemies",
+            {"or": [
+              "SpaceJump",
+              "canSpringBallJumpMidAir"
+            ]}
+          ]}
+        ]}
+      ]
+    },
+    {
+      "link": [1, 4],
+      "name": "Tank Damage",
+      "requires": [
+        "canCarefulJump",
+        {"enemyDamage": {
+          "enemy": "Tourian Space Pirate (all)",
+          "type": "contact",
+          "hits": 2
+        }},
+        {"or": [
+          "HiJump",
+          "canWalljump",
```

Comment:
I think, `canConsecutiveWalljump` for difficulty placement.  Or it would take more pirate hits to climb up, at medium.

Reply:
`canConsecutiveWalljump` is a Medium tech so it wouldn't change the difficulty. I think you're right it's a problem. Medium players will probably get knocked down almost indefinitely while trying to walljump past the Pirates. It needs some non-trivial enemy dodging to avoid that outcome, and this fits more in Hard. So seems best to just remove the `canWalljump` option here. The Hard-level Pirate evade with walljump is covered already in the other strat.